utils/kubernetes/deployment.go

```diff
@@ -0,0 +1,178 @@
+			// Deployment failed
+			return &reconcile.Result{}, err
+		} else {
+			// Deployment was successful
+			return nil, nil
+		}
+	} else if err != nil {
+		// Error that isn't due to the deployment not existing
+		return &reconcile.Result{}, err
+	} else {
+		// If the deployment exists already there is an update to do
+		updateErr := c.Update(context.TODO(), dep)
+		if err != nil {
+			// Error that isn't due to the deployment not existing
+			return &reconcile.Result{}, updateErr
+		} else {
+			// Deployment was successful
+			return nil, nil
+		}
+	}
+}
+
+// createDeployment is a code for Creating Deployment
+func createDeployment(scheme *runtime.Scheme, v *v08.KogitoServerlessWorkflow, registryAddress string) *appsv1.Deployment {
+
+	labels := labels(v, v.Name)
+	size := int32(1)
+	dep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v.Name,
+			Namespace: v.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &size,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Image:           registryAddress + "/" + v.Name + utils.GetWorkflowImageTag(v),
+						ImagePullPolicy: corev1.PullAlways,
+						Name:            v.Name,
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: 8080,
+							Name:          "http",
+						}},
+					}},
+				},
+			},
+		},
+	}
+
+	controllerutil.SetControllerReference(v, dep, scheme)
+	return dep
+}
+
+// EnsureService ensures Service is Running in a namespace.
+func EnsureService(c ctrl.Client,
+	scheme *runtime.Scheme,
+	instance *v08.KogitoServerlessWorkflow,
+) (*reconcile.Result, error) {
+	service := createService(scheme, instance)
+	// See if service already exists and create if it doesn't
+	found := &appsv1.Deployment{}
+	err := c.Get(context.TODO(), types.NamespacedName{
+		Name:      service.Name,
+		Namespace: instance.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+
+		// Create the service
+		err = c.Create(context.TODO(), service)
+
+		if err != nil {
+			// Service creation failed
+			return &reconcile.Result{}, err
+		} else {
+			// Service creation was successful
+			return nil, nil
+		}
+	} else if err != nil {
+		// Error that isn't due to the service not existing
+		return &reconcile.Result{}, err
+	}
+
+	return nil, nil
+}
+
+// createService is a code for creating a Service
+func createService(scheme *runtime.Scheme, v *v08.KogitoServerlessWorkflow) *corev1.Service {
+	labels := labels(v, v.Name)
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v.Name,
+			Namespace: v.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports: []corev1.ServicePort{{
+				Protocol:   corev1.ProtocolTCP,
+				Port:       80,
+				TargetPort: intstr.FromInt(8080),
+				NodePort:   30685,
```

Comment:
do we want node port as default?

Reply:
and, IIRC, when using node port, we should let the port selection in charge of the control plane.

Reply:
nope, it was a typo coming from a local test ;-)